### PR TITLE
Migrate Itinerary Checks to GraphQL

### PR DESCRIPTION
**Pull request by binh-dam-ibigroup**
### Checklist

- [ ] Appropriate branch selected _(all PRs must first be merged to `dev` before they can be merged to `master`)_
- [ ] Any modified or new methods or classes have helpful JavaDoc and code is thoroughly commented
- [ ] The description lists all applicable issues this PR seeks to resolve
- [ ] The description lists any configuration setting(s) that differ from the default settings
- [ ] All tests and CI builds passing

### Description

Requires ~#242~ and #219. 

This PR migrates itinerary checks to use GraphQL, so that existence checks are consistent with the trip search results.

### CODE REVIEW

**Review by JymDyerIBI — APPROVED** [review]
These are big changes, but they make sense to me (so far).

**Review by JymDyerIBI — APPROVED** [review]
Now I've looked over all the changes since I last reviewed them, they look good to me. 👍

File: src/main/resources/queries/planQuery.graphql
Comment (br648) [inline src/main/resources/queries/planQuery.graphql:1]: This query is massive! How brittle is it, if any changes are made to OTP is it going to break?

File: src/main/java/org/opentripplanner/middleware/utils/GraphQLUtils.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
+package org.opentripplanner.middleware.utils;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.*;
```
Comment (br648) [inline src/main/java/org/opentripplanner/middleware/utils/GraphQLUtils.java]: Nit. Expand to reference just required classes.

File: src/main/java/org/opentripplanner/middleware/otp/OtpDispatcher.java [hunk 2]
```diff
@@ -78,7 +79,23 @@ public static OtpDispatcherResponse sendOtpPlanRequest(OtpVersion version, Strin
...
+     */
+    public static OtpDispatcherResponse sendOtpPlanRequest(OtpVersion version, OtpGraphQLVariables params) {
+        OtpGraphQLQuery query = new OtpGraphQLQuery();
+        query.query = GraphQLUtils.getPlanQueryTemplate();
+        query.variables = params;
+        return sendOtpPostRequest(
+            version,
+            "",
+            OTP_GRAPHQL_ENDPOINT,
+            Map.of("Content-Type", "application/json"),
```
Comment (br648) [inline src/main/java/org/opentripplanner/middleware/otp/OtpDispatcher.java]: Perhaps use `HttpUtils.HEADERS_JSON`

File: src/main/java/org/opentripplanner/middleware/utils/ItineraryUtils.java [hunk 3]
```diff
@@ -31,38 +29,23 @@
  */
 public class ItineraryUtils {
 
-    public static final String IGNORE_REALTIME_UPDATES_PARAM = "ignoreRealtimeUpdates";
-    public static final String DATE_PARAM = "date";
     public static final String MODE_PARAM = "mode";
```
Comment (br648) [inline src/main/java/org/opentripplanner/middleware/utils/ItineraryUtils.java]: Doesn't appear to be used. Consider removing.

File: src/test/java/org/opentripplanner/middleware/utils/ItineraryUtilsTest.java
Comment (br648) [inline src/test/java/org/opentripplanner/middleware/utils/ItineraryUtilsTest.java:1]: These two tests fail for me:

testItineraryMatches
canCheckAllItinerariesExist

With the error:

com.fasterxml.jackson.databind.exc.MismatchedInputException: Cannot construct instance of `org.opentripplanner.middleware.otp.response.Route` (although at least one Creator exists): no String-argument constructor/factory method to deserialize from String value ('MAX Blue Line')
 at [Source: (StringReader); line: 621, column: 24] (through reference chain: org.opentripplanner.middleware.otp.response.OtpResponseGraphQLWrapper["data"]->org.opentripplanner.middleware.otp.response.OtpResponse["plan"]->org.opentripplanner.middleware.otp.response.TripPlan["itineraries"]->java.util.ArrayList[0]->org.opentripplanner.middleware.otp.response.Itinerary["legs"]->java.util.ArrayList[1]->org.opentripplanner.middleware.otp.response.Leg["route"])

**Review by br648 — CHANGES_REQUESTED** [review]
Some minor items to address. On the face of it this update is good. My only concern is maintaining planQuery.graphql. I guess if OTP is not updated this is fine and we don't need to worry?

File: src/test/java/org/opentripplanner/middleware/utils/ItineraryUtilsTest.java
Comment (binh-dam-ibigroup) [reply src/test/java/org/opentripplanner/middleware/utils/ItineraryUtilsTest.java:1]: Fixed in https://github.com/ibi-group/otp-middleware/pull/243/commits/3c120c7ed335e78bf600559ee29ec7fdb55472b5.

File: src/main/java/org/opentripplanner/middleware/utils/ItineraryUtils.java (hunk 3 above)
Comment (binh-dam-ibigroup) [reply src/main/java/org/opentripplanner/middleware/utils/ItineraryUtils.java]: Fixed in https://github.com/ibi-group/otp-middleware/pull/243/commits/340fc1f447e0d27adc41e5518d9229a8bdf0e061

File: src/main/java/org/opentripplanner/middleware/otp/OtpDispatcher.java (hunk 2 above)
Comment (binh-dam-ibigroup) [reply src/main/java/org/opentripplanner/middleware/otp/OtpDispatcher.java]: Fixed in https://github.com/ibi-group/otp-middleware/pull/243/commits/4ff727995490aec5780d108974d228c0267371e5

File: src/main/java/org/opentripplanner/middleware/utils/GraphQLUtils.java (hunk 1 above)
Comment (binh-dam-ibigroup) [reply src/main/java/org/opentripplanner/middleware/utils/GraphQLUtils.java]: Fixed in https://github.com/ibi-group/otp-middleware/pull/243/commits/4ff727995490aec5780d108974d228c0267371e5

**Review by br648 — APPROVED** [review]
I'm approving, but still have concerns over the maintainability of `planQuery.graphql` and the possible impact on performance on the OTP server being queried. Should I be concerned?!

**Pull request merged**